### PR TITLE
Database backup and documentation

### PR DESCRIPTION
**Pull request by gitkvark**

### CODE REVIEW

**Comment by JoAngel8** [comment]
Backup documentation **Draft version** (missings few points and backup plan explications)

File: docs/documentation/backup_plan.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+> 
+> Cette manipulation sera réalisée manuellement sur la console AWS, sélectionner le service IAM
+> * Nous devons créer un nouvel utilisateur "de service" IAM 
+> * Générer une clé d'accès pour cet utilisateur IAM
+>
+> Manipulation manuelle depuis la Console AWS, sélectionner le service S3
+> * Créer l'objet S3 bucket
+> * Autoriser le rôle et les accès à dédier à l'utilisateur IAM
+>
+> Cette étape ne doit pas être répétée. Il parait suffisant de le réaliser manuellement et de bien le documenter ainsi il sera aisé de reproduire le déploiement à l'identique.
```
Comment (silopolis) [inline docs/documentation/backup_plan.md:19]: I should be able to provide AWS CLI commands without much effort...

File: docs/documentation/backup_plan.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+
+### Prérequis
+
+Accéder au site (https://aws.amazon.com/fr/console/)
+se connecter en tant qu'utilisarteur IAM, avec notre compte (ID de compte, Nom d'utilisateur, mot de passe)
+Se positionner sur la région de
+* Paris pour gérer le service IAM
+* Californie pour le service S3. (limitation de notre compte IAM Datascientest)
```
Comment (silopolis) [inline docs/documentation/backup_plan.md:28]:
```suggestion
- Accéder au site (https://aws.amazon.com/fr/console/)
- Se connecter en tant qu'utilisarteur IAM, avec notre compte (ID de compte, Nom d'utilisateur, mot de passe)
- Se positionner sur la région de
  * Paris pour gérer le service IAM
  * Californie pour le service S3. (limitation de notre compte IAM Datascientest)
```

File: docs/documentation/backup_plan.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+### 1.2 Création d'un compte utilisateur
+
+
```
Comment (silopolis) [inline docs/documentation/backup_plan.md:32]:
```suggestion


### 1.2 Création d'un compte utilisateur

```

File: docs/documentation/backup_plan.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+Le rôle IAM permet l'accès au Service S3 pour stocker les fichiers de sauvegarde. 
+Ce rôle se différentie de notre accès IAM principale car il aura des privilèges limités d'une part car il aura **accès uniquement au Bucket S3** et d'autre part ses **permissions seront défines au stricte nécessaire** pour préserver les données en cas d'attaque.
```
Comment (silopolis) [inline docs/documentation/backup_plan.md:34]:
```suggestion

Ce rôle se différentie de notre accès IAM principal car il aura des privilèges limités car:
* il aura **accès uniquement au Bucket S3**,
* ses **permissions seront défines au stricte nécessaire** pour préserver les données en cas d'attaque.
```

File: docs/documentation/backup_plan.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+* Région Paris
+* Selectionner Gestion des accès / Utilisateurs
+* Bouton Créer un utilisateur
+  * Nommer l'utilisateur (le nom n'a pas d'importance) : eks-srvuser
+  * Pas d'accès console nécessaire, ne pas cocher.
+  * Sélectionner "Ajouter un utilisateur à un groupe", suivant
+  * Bouton "Créer un utilisateur"
+
+* Selectionner l'utilisateur créé
+  * **Relever la reférence ARN**
```
Comment (silopolis) [inline docs/documentation/backup_plan.md:48]:
```suggestion
#### Procédure

- Ouvrir le Console "Identity and Access Management (IAM)"
- Région Paris
- Selectionner Gestion des accès / Utilisateurs
- Bouton Créer un utilisateur

  * Nommer l'utilisateur (le nom n'a pas d'importance) : eks-srvuser
  * Pas d'accès console nécessaire, ne pas cocher.
  * Sélectionner "Ajouter un utilisateur à un groupe", suivant
  * Bouton "Créer un utilisateur"

- Selectionner l'utilisateur créé

  * **Relever la reférence ARN**

```

File: docs/documentation/backup_plan.md [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+### 1.3 Générer la clé d'accès du rôle IAM
+
+La clé d'accès du rôle IAM est la methode d'authentification utilisée afin de permettre  l'accès au Bucket S3. Cette information sera ensuite renseignée dans les manifestes helm postgres.
+
+**Procédure**
```
Comment (silopolis) [inline docs/documentation/backup_plan.md:54]:
```suggestion
#### Procédure
```

File: docs/documentation/backup_plan.md [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+
+* Poursuivons dans la même fenêtre
+  * selectionner l'onglet: "Information d'identification de sécurité"
+  * Bouton Créer une clé d'accès
+  * Cas d'utilisation: "Interface de ligne de commande (CLI)", cocher "Je comprends les recommandations...", suivant.
+  * Renseigner la description "Description de la valeur d'identification", bouton "Créer une clé d'accès"
+  * Important, bien **Récupérer les clés d'accès** (Clé d'accès et d'accès secrète)
+  * Bouton Terminer.
```
Comment (silopolis) [inline docs/documentation/backup_plan.md:62]:
```suggestion
Poursuivons dans la même fenêtre

- Selectionner l'onglet: "Information d'identification de sécurité"
- Bouton "Créer une clé d'accès"
- Cas d'utilisation: "Interface de ligne de commande (CLI)", cocher "Je comprends les recommandations...", suivant.
- Renseigner la description "Description de la valeur d'identification", bouton "Créer une clé d'accès"
- Important, bien **Récupérer les clés d'accès** (Clé d'accès et d'accès secrète)
- Bouton "Terminer".

```

File: docs/documentation/backup_plan.md [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+**Procédure**
+
+* Ouvrir le Console "Amazon S3"
+* Bouton "Créer un compartiment" ou "Bucket"
+* Région AWS, selectionner USA Est (Californie du Nord) us-east-1
+* Type: "usage général"
+* Nom du compartiement (bucket name): "devops-boot-s3backup", Atttention, il doit être unique dans l'espace de nommage global!
+* Le type de chiffrement par défaut (Objet lock): "Chiffrement côté serveur avec des clés gérées par Amazon S3 (SSE-S3)"
+* Section Clé de compartiement: "Désactivée"
+* Bouton: "Créer un compartiement"
```
Comment (silopolis) [inline docs/documentation/backup_plan.md:81]:
```suggestion
#### Procédure

- Ouvrir le Console "Amazon S3"
- Bouton "Créer un compartiment" ou "Bucket"
- Région AWS, selectionner USA Est (Californie du Nord) us-east-1
- Type: "usage général"
- Nom du compartiement (bucket name): "devops-boot-s3backup".

  Atttention, il doit être **unique** dans l'espace de nommage global!

- Le type de chiffrement par défaut (Objet lock): "Chiffrement côté serveur avec des clés gérées par Amazon S3 (SSE-S3)"
- Section Clé de compartiement: "Désactivée"
- Bouton: "Créer un compartiement"
```

File: docs/documentation/backup_plan.md [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+
+Une fois créé, afficher les détails du bucket "devops-boot-s3backup"
+
+* Onglet Propriété, **Relever la chaine ARN** (Amazon Ressource Name): "arn:aws:s3:::devops-boot-s3backup"
```
Comment (silopolis) [inline docs/documentation/backup_plan.md:85]:
```suggestion
Une fois créé, afficher les détails du bucket "devops-boot-s3backup", dans l'onglet "Propriétés", **relever la chaine ARN** (Amazon Ressource Name): "arn:aws:s3:::devops-boot-s3backup"
```

File: docs/documentation/backup_plan.md [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+* Nous avons utilisé le bouton "AWS Policy Generator" pour générer le code JSON standard.
+  * Step 1 - Policy Type selectionnée : S3 Bacuket Policy
+  * Step 2 - Add Statement(s):
+        - Effect : allow (Objectif est d’autoriser, permettre l’accès) 
+        - Principal : arn:aws:iam::424571028400:user/eks-srvuser (Notre utilisateur IAM role pour l’accès)
+        - Actions : All actions (temporaire)
+        - Amazon Ressource Name (ARN) : arn:aws:s3:::devops-boot-s3backup (Notre bucket fraichement créé)
+  * Step 3 - Generate Policy # Il s’agit de valider nos choix. Une seule règle est nécessaire dans notre cas.
+  * copier / Coller du code dans notre espace Bucket Policy.
+  * Bouton Save changes
```
Comment (silopolis) [inline docs/documentation/backup_plan.md:106]:
```suggestion
#### Procédure

Poursuivons dans la même fenêtre:

- Selectionner l'Onglet "Permission"
- Bucket Policy, "Editer"
- Lancer le "AWS Policy Generator" pour générer le code JSON standard:

  - Step 1 - Policy Type selectionnée : S3 Bacuket Policy
  - Step 2 - Add Statement(s):
    * Effect : allow (Objectif est d’autoriser, permettre l’accès) 
    * Principal : arn:aws:iam::424571028400:user/eks-srvuser (Notre utilisateur IAM role pour l’accès)
    * Actions : All actions (temporaire)
    * Amazon Ressource Name (ARN) : arn:aws:s3:::devops-boot-s3backup (Notre bucket fraichement créé)
  - Step 3 - "Generate Policy".
  
    Il s’agit de valider nos choix. Une seule règle est nécessaire dans notre cas.

  - Copier / Coller du code dans notre espace Bucket Policy.
  - Bouton "Save changes"
```

File: docs/documentation/backup_plan.md [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+
+Maintenant, nous avons autorisé au principal de pouvoir accéder au Bucket. Notre utilisateur a maintenant accès au Bucket.
```
Comment (silopolis) [inline docs/documentation/backup_plan.md:108]:
```suggestion
Désormais, notre utilisateur ("principal") a accès au Bucket.
```

File: docs/documentation/backup_plan.md [hunk 12]
```diff
@@ -0,0 +1,189 @@
...
+
+> Contrainte, il est nécessaire d'adapter avec nos besoins (niveau d'accès, quelle ressources) et nos données spécifiques en l'occurence les informations que nous avons relevés plus tôt dans notre démarche.
```
Comment (silopolis) [inline docs/documentation/backup_plan.md:110]:
```suggestion
Il est maintenant nécessaire d'adapter la politique générée à nos besoins (niveau d'accès, ressources concernées) et nos données spécifiques, en l'occurence, les informations que nous avons relevées plus tôt.

```

File: docs/documentation/backup_plan.md [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+                "arn:aws:s3:::devops-boot-s3backup",     # Bucket S3 path
+                "arn:aws:s3:::devops-boot-s3backup/*"    # Bucket S3 level access.
+            ]
+        }
+    ]
+}
+```
+***Règle JSON personnalisée attribuée au compartiement (JSON Bucket policy)***
+
+### 2.1 - Installer et utiliser K9S pour superviser Kubernetes
```
Comment (silopolis) [inline docs/documentation/backup_plan.md:134]:
```suggestion

### 2.1 - Installer et utiliser K9S pour superviser Kubernetes
```

File: docs/documentation/backup_plan.md [hunk 14]
```diff
@@ -0,0 +1,189 @@
...
+
+**A developper**
```
Comment (silopolis) [inline docs/documentation/backup_plan.md]:
```suggestion
<!-- TODO Split out and expand -->
```

File: docs/documentation/backup_plan.md [hunk 15]
```diff
@@ -0,0 +1,189 @@
...
+* configurer les variables d'environnement: AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY
+* Récupérer le cluster config:
+```
+aws eks update-kubeconfig --region eu-west-3 --name [cluster_name] --kubeconfig ~/.kube/[cluster_name].config
+```
+* Configure le variable KUBECONFIG: 
+```
+export KUBECONFIG=~/.kube/[cluster_name].config
+k9s
+```
```
Comment (silopolis) [inline docs/documentation/backup_plan.md:151]:
```suggestion
```console
sudo apt-get install snapd
sudo snap install k9s
```

* Configurer les variables d'environnement: `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`

* Récupérer le cluster config:

```console
aws eks update-kubeconfig --region eu-west-3 --name [cluster_name] --kubeconfig ~/.kube/[cluster_name].config
```

* Configurer la variable `KUBECONFIG`: 

```console
export KUBECONFIG=~/.kube/[cluster_name].config
k9s
```

```

File: docs/documentation/backup_plan.md [hunk 16]
```diff
@@ -0,0 +1,189 @@
...
+(helm/postgres/templates/postgres.yaml)
+
+**Contraintes**
+
+* L'opérateur PostGresSQL doit être dans le namespace default. Sinon, un paramétrage complémentaire RBAC est nécessaire.
+  
+* Nous avons pu observer que en cas d'erreur l'outil à réaliser une restauration automatiquement. Il faudra inclure un controle d'intégrité de la sauvegarde ?
+
+**Procédure**
+
```
Comment (silopolis) [inline docs/documentation/backup_plan.md:174]:
```suggestion
#### Contraintes

* L'opérateur PostGresSQL doit être dans le namespace default.

  Sinon, un paramétrage complémentaire RBAC est nécessaire.
  
* Nous avons pu observer que, en cas d'erreur, l'outil effectue une restauration automatiquement.

  Il faudra inclure un controle d'intégrité de la sauvegarde ?

#### Procédure
```

**Review by silopolis — APPROVED** [review]
Well well, fantastic work as always! :sunglasses: 

Reviewed documentation to my best. That's mostly formatting and Markdown good practices fixes(numbered lists for procedures, no bold/em instead of headings, blank lines around headings, etc.), and a tiny bit of wording adaptations.

Thanks for this one again

File: docs/documentation/backup_plan.md [hunk 17]
```diff
@@ -0,0 +1,189 @@
+Plan de sauvegarde
+=======
+---
```
Comment (silopolis) [inline docs/documentation/backup_plan.md:3]:
```suggestion
# Plan de sauvegarde

```

File: docs/documentation/backup_plan.md (hunk 1 above)
Comment (gitkvark) [reply docs/documentation/backup_plan.md:50]: We can add them to the doc when you have them.

File: docs/documentation/backup_plan.md (hunk 14 above)
Comment (JoAngel8) [reply docs/documentation/backup_plan.md]: ok

**Review by JoAngel8 — APPROVED** [review]
review done.

**Pull request merged**